app/controllers/users_controller.rb

```diff
@@ -0,0 +1,16 @@
+class UsersController < ApplicationController
+  def create
+    conn = Faraday.new(url: 'http://localhost:3000/')
+
+    auth_hash = request.env['omniauth.auth']
+    email = auth_hash['info']['email']
+
+    session[:access_token] = auth_hash['credentials']['token']
+
+    response = conn.post('/api/v1/users') do |req|
+      req.headers['Content-Type'] = 'application/json'
+      req.body = JSON.generate(user: { username: email, email: email })
+    end
+    redirect_to '/'
```

Comment:
This should probably redirect to dashboard

Reply:
Agreed